examples/3d/edit_material_on_gltf.rs

```diff
@@ -0,0 +1,95 @@
+    gltf::GltfAssetLabel,
+    math::{Dir3, Vec3},
+    pbr::{DirectionalLight, MeshMaterial3d, StandardMaterial},
+    prelude::{Camera3d, Children, Commands, Component, Query, Res, ResMut, Transform, Trigger},
+    scene::{SceneInstanceReady, SceneRoot},
+    DefaultPlugins,
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins.build().disable::<AudioPlugin>())
+        .add_systems(Startup, setup_scene)
+        .add_observer(change_material)
+        .run();
+}
+
+/// This is added to a [`SceneRoot`] and will cause the [`StandardMaterial::base_color`]
+/// of all materials to be overwritten
+#[derive(Component)]
+struct ColorOverride(Color);
+
+fn setup_scene(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands.spawn((
+        Camera3d::default(),
+        Transform::from_xyz(0., 1., 2.5).looking_at(Vec3::new(0., 0.25, 0.), Dir3::Y),
+    ));
+
+    commands.spawn((
+        DirectionalLight::default(),
+        Transform::from_xyz(0., 1., 0.25).looking_at(Vec3::ZERO, Dir3::Y),
+    ));
+
+    // This model will keep its original materials
+    commands.spawn(SceneRoot(asset_server.load(
+        GltfAssetLabel::Scene(0).from_asset("models/FlightHelmet/FlightHelmet.gltf"),
+    )));
+    // This model will be tinted red
+    commands.spawn((
+        SceneRoot(
+            asset_server
+                .load(GltfAssetLabel::Scene(0).from_asset("models/FlightHelmet/FlightHelmet.gltf")),
+        ),
+        Transform::from_xyz(-1.25, 0., 0.),
+        ColorOverride(palettes::tailwind::RED_300.into()),
+    ));
+    // This model will be tinted green
+    commands.spawn((
+        SceneRoot(
+            asset_server
+                .load(GltfAssetLabel::Scene(0).from_asset("models/FlightHelmet/FlightHelmet.gltf")),
+        ),
+        Transform::from_xyz(1.25, 0., 0.),
+        ColorOverride(palettes::tailwind::GREEN_300.into()),
+    ));
+}
+
+fn change_material(
+    trigger: Trigger<SceneInstanceReady>,
+    mut commands: Commands,
+    children: Query<&Children>,
+    color_override: Query<&ColorOverride>,
+    mesh_materials: Query<&MeshMaterial3d<StandardMaterial>>,
+    mut asset_materials: ResMut<Assets<StandardMaterial>>,
+) {
+    // Get the `ColorOverride` of the entity, if it does not have a color override, skip
+    let Ok(color_override) = color_override.get(trigger.target()) else {
+        return;
+    };
+
+    // Iterate over all children recursively
+    for descendants in children.iter_descendants(trigger.target()) {
+        // Get the material of the descendant
+        if let Some(material) = mesh_materials
+            .get(descendants)
+            .ok()
+            .and_then(|id| asset_materials.get_mut(id.id()))
+        {
+            // Create a copy of the material and override base color
+            let mut new_material = material.clone();
+            new_material.base_color = color_override.0;
+
+            // Override `MeshMaterial3d` with new material
```

Comment:
Re-surfacing this comment from a now-outdated review:

I still think that the act of adding multiple brand new materials every time this observer fires is an important enough detail to call out more explicitly in a comment.
 
This is a silent performance footgun, and users likely want to find a way to reuse the materials. I feel like actually "fixing" this may be advanced / context dependent so a comment should suffice.